dot/network/stream_manager.go

```diff
@@ -23,14 +23,15 @@ type streamData struct {
 // This prevents keeping stale streams open and continuously trying to
 // read from it, which takes up lots of CPU over time.
 type streamManager struct {
-	ctx           context.Context
-	streamDataMap *sync.Map //map[string]*streamData
+	ctx          context.Context
+	streamDataMu sync.RWMutex
```

Comment:
should it just be `sync.Mutex` since `RLock` isn't used?

Reply:
Totally, done! :+1: